eserde_derive/src/model.rs

```diff
@@ -70,81 +98,112 @@ impl PermissiveCompanionType {
                             }
                         }
                     }
-                    if add_borrow_attr && !has_serde_path_attr(&field.attrs, "borrow") {
-                        field.attrs.push(syn::parse_quote!(#[serde(borrow)]));
+                    if add_borrow_attr && find_attr_meta(&field.attrs, "serde", "borrow").is_none()
+                    {
+                        field
+                            .attrs
+                            .push(syn::parse_quote_spanned!(span=> #[serde(borrow)]));
                     }
                 }
 
                 // Check if `#[serde(default)]` is already present on the field.
                 // TODO: handle the `#[serde(default = "..")]` case.
                 //   We'll have to generate a function that wraps around the
                 //   one specified in the attribute.
```

Comment:
The `default = ".."` handling is a lot easier than this to fix actually, just clear the ` = ".."` and continue as normal